framework/include/materials/MaterialPropertyStorage.h

```diff
@@ -190,45 +192,63 @@ class MaterialPropertyStorage
    */
   bool hasOlderProperties() const { return _has_older_prop; }
 
+  /**
+   * Accessible type of the stored material property data.
+   *
+   * This probably should have been returned as a proxy class; only
+   * access it via foo[elem][side] and maybe we'll be able to refactor
+   * it in the future without breaking your code.
+   */
+  typedef HashMap<const Elem *, HashMap<unsigned int, MaterialProperties>> PropsType;
+
   ///@{
   /**
-   * Access methods to the stored material property data
+   * Const access methods to the stored material property data.
    *
+   * We used to have non-const accessors too, but the violation of
+   * encapsulation there was too much.
    */
-  HashMap<const Elem *, HashMap<unsigned int, MaterialProperties>> & props()
-  {
-    return *_props_elem;
-  }
-  HashMap<const Elem *, HashMap<unsigned int, MaterialProperties>> & propsOld()
-  {
-    return *_props_elem_old;
-  }
-  HashMap<const Elem *, HashMap<unsigned int, MaterialProperties>> & propsOlder()
-  {
-    return *_props_elem_older;
-  }
-  const HashMap<const Elem *, HashMap<unsigned int, MaterialProperties>> & props() const
+  const PropsType & props() const { return *_props_elem; }
+  const PropsType & propsOld() const { return *_props_elem_old; }
+  const PropsType & propsOlder() const { return *_props_elem_older; }
+  const MaterialProperties & props(const Elem * elem, unsigned int side) const
   {
-    return *_props_elem;
+    libmesh_assert(props().contains(elem));
+    libmesh_assert(props().find(elem)->second.contains(side));
+    return props().find(elem)->second.find(side)->second;
   }
```

Comment:
Maybe take these multi-line methods out of the class definition? Can inline them in the header outside the class definition